share-coordinator/src/main/java/org/apache/kafka/coordinator/share/ShareCoordinator.java

```diff
@@ -32,11 +32,6 @@
 import java.util.function.IntSupplier;
 
 public interface ShareCoordinator {
-    short SHARE_SNAPSHOT_RECORD_KEY_VERSION = 0;
-    short SHARE_SNAPSHOT_RECORD_VALUE_VERSION = 0;
-    short SHARE_UPDATE_RECORD_KEY_VERSION = 1;
-    short SHARE_UPDATE_RECORD_VALUE_VERSION = 1;
```

Comment:
This is a bug. The version of the update record must be 0.

Reply:
@AndrewJSchofield Is this something that we want to fix in 4.0?

Reply:
Pardon me, why not fix it for 4.0? I assume it would be an issue to 4.1 in reading the records having incorrect version.